### PR TITLE
Fix segfaults

### PR DESCRIPTION
**Pull request by huyang1988**

### CODE REVIEW

**Comment by amygara** [comment]
Is this the final product for demarshal_client?

File: lib/marshal/marshal.c [hunk 1]
```diff
@@ -189,46 +191,51 @@ gwkv_demarshal_client(char* ascii, struct operation** op, int* status)
...
                 i++;
         }       
         b[i]='\0';
         offset += i+1;
         char c1[]="VALUE";
-        if(strcmp(b,c1)!=0){
-                return -1;
-        }
+printf("b= %s\n",b);
+//        if(strcmp(b,c1)!=0){
```
Comment (amygara) [inline lib/marshal/marshal.c:209]: For everything commented like this, are these needed or should they be removed?

**Comment by amygara** [comment]
@phil-lopreiato @huyang1988  Did #55 fix this and should this be closed?

**Comment by phil-lopreiato** [comment]
#55 doesn't segfault and all the tests pass, so I think we don't need this anymore. @huyang1988 can you confirm?

**Comment by pradeep-k** [comment]
@phil-lopreiato : I wrote some additional testcases yesterday and changed the code in the APIs so the snapshot of the code is no more valid. So, I believe that has fixed the issue. Raise a new issue if you find any further bugs and tag me.

**Comment by phil-lopreiato** [comment]
Sounds good, closing this

**Pull request closed without merging**